app/generic/proof/facades/acapy_proof_v1.py

```diff
@@ -0,0 +1,180 @@
+from typing import Dict, Optional, Union
+import logging
+
+from aries_cloudcontroller import (
+    AcaPyClient,
+    AdminAPIMessageTracing,
+    IndyPresSpec,
+    IndyProofRequest,
+    V10PresentationCreateRequestRequest,
+    V10PresentationProblemReportRequest,
+    V10PresentationProposalRequest,
+    V10PresentationSendRequestRequest,
+)
+from aries_cloudcontroller.model.v10_presentation_exchange import (
+    V10PresentationExchange,
+)
+from fastapi.exceptions import HTTPException
+from pydantic.typing import NoneType
+
+from app.generic.proof.facades.acapy_proof import Proof
+from app.generic.proof.models import PresentationExchange
+
+logger = logging.getLogger(__name__)
+
+
+class ProofsV1(Proof):
+    @classmethod
+    async def create_proof_request(
+        cls,
+        controller: AcaPyClient,
+        proof_request: IndyProofRequest,
+        comment: str = None,
+        trace: bool = False,
+    ) -> V10PresentationExchange:
+        presentation_exchange = (
+            await controller.present_proof_v1_0.create_proof_request(
+                body=V10PresentationCreateRequestRequest(
+                    proof_request=proof_request,
+                    comment=comment,
+                    trace=trace,
+                )
+            )
+        )
+        return cls.__record_to_model(presentation_exchange)
+
+    @classmethod
+    async def send_proof_request(
+        cls,
+        controller: AcaPyClient,
+        proof_request: Union[
+            V10PresentationSendRequestRequest,
+            AdminAPIMessageTracing,
+            V10PresentationProposalRequest,
+        ],
+        free: bool = False,
+        pres_ex_id: str = None,
+    ) -> PresentationExchange:
+        try:
+            # This "free" is de facto the only one we support right now
+            if free:
+                presentation_exchange = (
+                    await controller.present_proof_v1_0.send_request_free(
+                        body=V10PresentationSendRequestRequest(**proof_request.dict())
+                    )
+                )
+            elif isinstance(proof_request, AdminAPIMessageTracing) and pres_ex_id:
+                presentation_exchange = (
+                    await controller.present_proof_v1_0.send_request(
+                        pres_ex_id=pres_ex_id, body=proof_request
+                    )
+                )
```

Comment:
As I said in the other comment, I would ignore tracing for now. This way there is only one case to handle (the first one)